src/persist/benches/writer.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Benchmarks for different persistent Write implementations.
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use criterion::{criterion_group, criterion_main, Bencher, Criterion};
+
+use persist::file::FileStream;
+use persist::mem::MemStream;
+use persist::persister::Write;
+
+fn bench_write_sync<W: Write + Clone>(
+    writer: &W,
+    data: &[((String, String), u64, isize)],
+    b: &mut Bencher,
+) {
+    b.iter_with_setup(
+        || writer.clone(),
+        |mut writer| writer.write_sync(&data).expect("failed to write data"),
+    )
+}
+
+pub fn bench_writes(c: &mut Criterion) {
+    let data = vec![
+        (("key1".to_string(), "val1".to_string()), 1, 1),
+        (("key2".to_string(), "val2".to_string()), 1, 1),
+    ];
+
+    let mem_stream = MemStream::new();
+    c.bench_function("mem_write_sync", |b| {
+        bench_write_sync(&mem_stream, &data, b)
+    });
+
+    // Create a directory that will automatically be dropped after the test finishes.
+    let temp_dir = tempfile::tempdir().expect("failed to create temp directory");
```

Comment:
I don't love having to import a tempfile and manually name it here again - its duplication of the unit test code. But this is what testdrive seems to do and a cursor look at the tempfile docs didn't show a better way to do things given our interface (FileStream wants a path, which i think makes sense, tempfile can give you a File, which also makes sense but doesnt help us here)

Reply:
i'm not worried about it. people tend to focus far too much on DRY, often to the actual detriment of the code

anyway, I think `tempfile::NamedTempFile()` might be what you want? that would let us get rid of the systemtime stuff